tools/Microsoft.StandardsPolice/StandardsPoliceCompileModule.cs

```diff
@@ -0,0 +1,288 @@
+﻿using Microsoft.Dnx.Compilation.CSharp;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp;
+using System;
+
+namespace Microsoft.StandardsPolice
+{
+    public class StandardsPoliceCompileModule : ICompileModule
+    {
+        public void BeforeCompile(BeforeCompileContext context)
+        {
+            ScanCompilation(context.Diagnostics, context.Compilation);
+        }
+
+        internal static void ScanCompilation(IList<Diagnostic> diagnostics, CSharpCompilation compilation)
+        {
+            ScanNamespace(diagnostics, compilation.GlobalNamespace);
+
+            foreach (var st in compilation.SyntaxTrees)
+            {
+                if (!st.FilePath.EndsWith(".Generated.cs"))
```

Comment:
Because there are more than one type in that file, and it would be harder to change the generation to write multiple files